services/api/service.go

```diff
@@ -286,6 +298,9 @@ func (api *RelayAPI) StartServer() (err error) {
 	if api.opts.BlockBuilderAPI {
 		// Get current proposer duties blocking before starting, to have them ready
 		api.updateProposerDuties(bestSyncStatus.HeadSlot)
+
+		// TODO(mikeneuder): consider if we should use >1 optimisticBlockProcessors.
+		go api.startOptimisticBlockProcessor()
```

Comment:
`optimisticBlockProcessors` are extremely lightweight, right? Why have more than 1?

Reply:
i guess it depends how many optimistic blocks we get. `startOptimisticBlockProcessor` waits for the response of the block simulation from the prio-load-balancer, so if we only have one goroutine running this we are processing all the optimistic blocks serially. i thought it might be desirable to have a few goroutines sending the simulation requests because we have multiple validation nodes running. thoughts?

Reply:
> it depends how many optimistic blocks we get

We should expect to get hammered with optimistic blocks, partly because all the builders will want to connect to the ultra sound relay, but also partly because we can relax limit QoS limits (like max 2 blocks/sec).

>  if we only have one goroutine running this we are processing all the optimistic blocks serially

My naive understanding was that `startOptimisticBlockProcessor` calls `simulateBlock`, which calls `api.blockSimRateLimiter.send`, which itself uses parallelism.

Reply:
that is the call path where we call api.BlockSimRateLimiter.send. that function can handle a bunch of concurrent blocks, but it still waits on the response from the simulation for the single block that we sent. so if we only have a single `go startOptimisticBlockProcessor()` I think we are only processing one block from that channel at a time. 

e.g., 
1. we receive a block from the channel
2. we call `api.BlockSimRateLimiter.send`
3. we wait until send returns to get the simulation error

whereas we could/should have multiple go routines listening on the channel, and concurrent calls to `api.BlockSimRateLimiter.send`. maybe i am missing something though.

Reply:
> concurrent calls to `api.BlockSimRateLimiter.send`

Ok, agreed we should have concurrent calls to `api.BlockSimRateLimiter.send` :) I don't have a particularly informed opinion as to the most natural way to do it here.